exporter/opentelemetry-exporter-otlp-proto-http/src/opentelemetry/exporter/otlp/proto/http/trace_exporter/encoder/__init__.py

```diff
@@ -1,288 +0,0 @@
-# Copyright The OpenTelemetry Authors
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-import logging
-from collections import abc
-from typing import Any, List, Optional, Sequence
-
-from opentelemetry.proto.collector.trace.v1.trace_service_pb2 import (
-    ExportTraceServiceRequest as PB2ExportTraceServiceRequest,
-)
-from opentelemetry.proto.common.v1.common_pb2 import AnyValue as PB2AnyValue
-from opentelemetry.proto.common.v1.common_pb2 import (
-    ArrayValue as PB2ArrayValue,
-)
-from opentelemetry.proto.common.v1.common_pb2 import (
-    InstrumentationScope as PB2InstrumentationScope,
-)
-from opentelemetry.proto.common.v1.common_pb2 import KeyValue as PB2KeyValue
-from opentelemetry.proto.resource.v1.resource_pb2 import (
-    Resource as PB2Resource,
-)
-from opentelemetry.proto.trace.v1.trace_pb2 import (
-    ScopeSpans as PB2ScopeSpans,
-)
-from opentelemetry.proto.trace.v1.trace_pb2 import (
-    ResourceSpans as PB2ResourceSpans,
```

Comment:
I don't know enough about semver. We kept all our code private in this module but if someone uses any tansient dep from this will it be a breaking change?

Reply:
Yep, this must be one I missed on adding back in with noqa 😞 I'll add it back in

Reply:
But this seems like a very weird use, I just don't see how you would end up using this transitive import, nothing is a public symbol in this module, apart from the imports, so you would be targeting the imports, a bit odd. This then makes me think do we have to add `from opentelemetry.... import Name as _Name` everywhere? Add this to any new addition?

Reply:
They could have something like this. Nothing prevents them from doing it but I am not sure if that comes under semvar / backward compatibility guarantee. 
```py
from opentelemetry.exporter.otlp.proto.http.trace_exporter.encoder import PB2ResourceSpans
```

What I have seen some projects do to avoid this kind of pitfall is every module will be private i.e module starts with `_` and only relevant APIs will be made public in `__init__.py` using `__all__`.

We followed this for our metrics SDK implementation https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-sdk/src/opentelemetry/sdk/metrics. There won't be such issues but there are some predated packages where this was not thought about.

Reply:
Yes, I would suggest doing that.